commons/all/src/test/java/org/infinispan/commons/configuration/io/yaml/YamlConfigurationReaderTest.java

```diff
@@ -171,4 +174,21 @@ private void assertAttribute(YamlConfigurationReader yaml, String name, String v
       }
       fail("Could not find attribute '" + name + " in element '" + yaml.getLocalName() + "'");
    }
+
+   @Test
+   public void testYamlMapper() throws IOException {
+      try (Reader r = new InputStreamReader(YamlConfigurationReaderTest.class.getResourceAsStream("/identities.yaml"))) {
+         YamlConfigurationReader yaml = new YamlConfigurationReader(r, new URLConfigurationResourceResolver(null), new Properties(), PropertyReplacer.DEFAULT, NamingStrategy.KEBAB_CASE);
+         Map<String, Object> identities = yaml.asMap();
+         assertEquals(1, identities.size());
+         List<Object> credentials = (List<Object>) identities.get("credentials");
+         assertNotNull(credentials);
+         assertEquals(3, credentials.size());
+         for (Object o : credentials) {
+            Map<String, Object> credential = (Map<String, Object>) o;
+            assertTrue(credential.containsKey("username"));
+            assertTrue(credential.containsKey("password"));
+         }
```

Comment:
nitpick, for completeness
```suggestion
                  for (Object o : credentials) {
            Map<String, Object> credential = (Map<String, Object>) o;
            assertTrue(credential.containsKey("username"));
            assertTrue(credential.containsKey("password"));
            assertEquals("changeme", credential.get("password"));
            if ("my-user-1".equals(credential.get("username"))) {
               assertTrue(credential.containsKey("roles"));
               assertEquals(Collections.singletonList("admin"), credential.get("roles"));
            } else if ("my-user-2".equals(credential.get("username"))) {
               assertTrue(credential.containsKey("roles"));
               assertEquals(Collections.singletonList("monitor"), credential.get("roles"));
            } else {
               assertEquals("admin", credential.get("username"));
               assertFalse(credential.containsKey("roles"));
            }
         }
```